subcommands/common.go

```diff
@@ -132,6 +134,20 @@ func Tabby(indent int, columns ...interface{}) *tabby.Tabby {
 	return tab
 }
 
+// Copied from canonical.MarshalIndent, but replaced the Marshal call with MarshalCanonical.
+func MarshalIndent(v interface{}, prefix, indent string) ([]byte, error) {
```

Comment:
There is a bug in canonical.MarshalIndent. I forget the exact issue, but I think it happens if you ever call it more than once. I *think* I hit it while playing on this commit:

 https://github.com/doanac/fioctl/commit/2b0256c65d1d49710fa8ab4dd45e63b205be04a1

and thus - still had to use `json.Marshal` to get the correct content

Reply:
I suspect that due to the _premature optimizations_ here https://github.com/docker/go/blob/master/canonical/json/encode.go#L264 and here https://github.com/docker/go/blob/master/canonical/json/encode.go#L332

On practice, after this commit there is only one place where we call `canonical.Marshal` twice: `fioctl keys resign-root`.
I tested it and it produces the correct result.

IIRC the problem appeared when using a mix of `canonical.MarshalCanonical` and `canonical.MarshalIndent` inter-changeably. But I rewrote our version of `MarshalIndent` in such a way that it is always _canonical_. So, as long as we never use `canonical/json` to produce non-canonical output - we should be fine.